app/routes/sprint-5/caseworkerManageRoutes.js

```diff
@@ -247,26 +256,55 @@ router.get("/referrals/:referralIndex/interventions/:interventionIndex/fast-forw
     res.redirect(`/sprint-5/book-and-manage/manage-a-referral/caseworker/referrals/${req.params.referralIndex}/interventions/${req.params.interventionIndex}`);
 });
 
-router.get("/referrals/:referralIndex/interventions/:interventionIndex/end-of-service-report", (req, res) => {
+router.get(`/referrals/:referralIndex/interventions/:interventionIndex/end-of-service-report`, (req, res) => {
     const intervention = findIntervention(req);
     const referral = findReferral(req);
 
-    res.render("sprint-5/book-and-manage/manage-a-referral/caseworker/end-of-service-report", { referralIndex: req.params.referralIndex, interventionIndex: req.params.interventionIndex, intervention, referral });
+    const isTermination = req.query.termination === "true";
+
+    res.render(`sprint-5/book-and-manage/manage-a-referral/caseworker/end-of-service-report`, { intervention, referralIndex: req.params.referralIndex, interventionIndex: req.params.interventionIndex, referral, isTermination });
 });
 
 router.post("/referrals/:referralIndex/interventions/:interventionIndex/end-of-service-report-check-answers", (req, res) => {
     const intervention = findIntervention(req);
     const referral = findReferral(req);
 
+    const isTermination = req.body.terminated === "true";
+
     endOfServiceReport = req.body;
 
-    res.render("sprint-5/book-and-manage/manage-a-referral/caseworker/end-of-service-report-check-answers", { referralIndex: req.params.referralIndex, interventionIndex: req.params.interventionIndex, intervention, referral, endOfServiceReport });
+    res.render("sprint-5/book-and-manage/manage-a-referral/caseworker/end-of-service-report-check-answers", { referralIndex: req.params.referralIndex, interventionIndex: req.params.interventionIndex, intervention, referral, endOfServiceReport, isTermination });
+});
+
+router.post("/referrals/:referralIndex/interventions/:interventionIndex/end-of-service-report-reason", (req, res) => {
+    switch (req.body.reason) {
+	case "delivered-successfully":
+	    res.redirect(`/sprint-5/book-and-manage/manage-a-referral/caseworker/referrals/${req.params.referralIndex}/interventions/${req.params.interventionIndex}/end-of-service-report`);
+	    break;
+	case "termination":
+	    res.redirect(`/sprint-5/book-and-manage/manage-a-referral/caseworker/referrals/${req.params.referralIndex}/interventions/${req.params.interventionIndex}/end-of-service-report-contacted-probation-practitioner`);
+	    break;
+    }
+});
+
+router.post("/referrals/:referralIndex/interventions/:interventionIndex/end-of-service-report-contacted-probation-practitioner", (req, res) => {
+    switch (req.body["contacted-probation-practitioner"]) {
+	case "yes":
```

Comment:
Is there any value in adding a "default" to this switch statement (and the one above?) or making the radio buttons required? It's not that likely, but if a user clicks "continue" without picking an option, where will they go?

Reply:
Yeah, good point, it'd just not send a response, which would be a Bad Thing to have to deal with in a research session. I don't really want to start introducing validation into the prototype, so I'll just make it redirect to the page with the question.